CHANGELOG.md

```diff
@@ -70,6 +70,42 @@ $govuk-suppressed-warnings: (
 
 This was added in [#2911 Add warning suppression functionality](https://github.com/alphagov/govuk-frontend/pull/2911)
 
+#### Configure components in JavaScript
+
+JavaScript components receiving configuration through data attributes now accept the same configuration options when instantiated in JavaScript. These components are:
+
+- the `Button` component, for its `preventDoubleClick` option (matching `data-prevent-double-click`)
+- the `CharacterCount` component, for its `maxlength`,`maxwords` and `threshold` options (matching `data-maxlength`, `data-maxwords` and `data-threshold`, respectively)
+- the `ErrorSummary` component, for its `disableAutoFocus` option (matching `data-disable-auto-focus`)
+- the `NotificationBanner` component, for its `disableAutoFocus` option (matching `data-disable-auto-focus`)
+
+You can omit these options when calling the Nunjucks macro and provide them:
+
+- at component instantiation, in a configuration object as second argument, for example:
+
+```js
+var errorSummary = document.querySelector('[data-module="error-summary"]')
+new GOVUKFrontend.ErrorSummary(errorSummary, {disableAutoFocus: true})
+```
+
+- when initialising components in bulk using `initAll`, for example:
+
+```js
+GOVUKFrontend.initAll({
+  errorSummary: {
+    disableAutoFocus: true
+  }
+  // Or, for the other components,
+  // button: {/* options */},
+  // characterCount: {/* options */},
+  // notificationBanner: {/* options */}
+})
+```
+
+If you provide an option both through data attribute and JavaScript configuration, the components will use the value from the data attribute.
```

Comment:
We might be able to hand off to the frontend docs for this level of detail

(It'd make sense to include a link either way)